patches/server/1053-Add-PlayerLandEvent.patch

```diff
@@ -0,0 +1,34 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: StillLutto <iameinsteinhe@gmail.com>
+Date: Sat, 8 Jun 2024 19:44:50 +0200
+Subject: [PATCH] Add PlayerLandEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+index 4ae88bfcead40cd05f9514a48a922a37767cb3cf..8e6d05933857764ef582a31088f7a97eed8c4cc9 100644
+--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
++++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+@@ -193,6 +193,7 @@ import net.minecraft.world.phys.Vec3;
+ import net.minecraft.world.phys.shapes.BooleanOp;
+ import net.minecraft.world.phys.shapes.Shapes;
+ import net.minecraft.world.phys.shapes.VoxelShape;
++import org.bukkit.craftbukkit.block.CraftBlock;
+ import org.slf4j.Logger;
+ 
+ // CraftBukkit start
+@@ -1591,6 +1592,15 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
+                                 this.player.absMoveTo(d0, d1, d2, f, f1);
+                                 boolean flag4 = this.player.isAutoSpinAttack();
+ 
++                                // Paper start - Add PlayerLandEvent
++                                if (flag2 && !(this.player.getY() - d4 >= 0.0)) {
++                                    this.player.setOnGround(true);
```

Comment:
What is up with setting the player's ground value here? As this is technically changing behavior?

Reply:
The reason I set the ground value here is because that method runs `checkSupportingBlock()` which updates `Entity#mainSupportingBlockPos`. The reason I'm not just taking advantage of `this.player.setOnGroundWithKnownMovement()` which happens just a few lines below is because I previously made this event cancellable, But after further testing, it had pretty buggy behaviour... I'm going to quickly just move this below the `this.player.setOnGroundWithKnownMovement()` in the next commit.